modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MuleExtensionUtils.java

```diff
@@ -298,6 +299,21 @@ public static Method getImplementingMethod(OperationDeclaration operationDeclara
         return operationDeclaration.getModelProperty(ImplementingMethodModelProperty.class).map(ImplementingMethodModelProperty::getMethod).orElse(null);
     }
 
+    /**
+     * If the {@code extensionModel} contains a {@link ClassLoaderModelProperty}, then it returns
+     * the {@link ClassLoader} associated to such property. If the property is not present,
+     * it returns the current {@link Thread}'s context classLoader.
+     *
+     * @param extensionModel a {@link ExtensionModel}
+     * @return a {@link ClassLoader}
+     */
+    public static ClassLoader getClassLoader(ExtensionModel extensionModel)
+    {
+        return extensionModel.getModelProperty(ClassLoaderModelProperty.class)
+                .map(ClassLoaderModelProperty::getClassLoader)
+                .orElse(Thread.currentThread().getContextClassLoader());
```

Comment:
Not sure that is correct to default to the Thread's context classLoader here, because if you do this, there is no reason to define the ClassLoaderModelProperty. I would use ExtensionModel.class.classloader instead, so if something depends on extension's code, then it will fail and we will know that there is an error because the ClassLoaderModelProperty is not processed correctly.